src/GitVersion.Core/Configuration/GitVersionConfiguration.cs

```diff
@@ -4,103 +4,107 @@
     [JsonPropertyPattern(@"'\+semver:\s?(feature|minor)'")]
-    public string? MinorVersionBumpMessage { get; set; }
+    public string? MinorVersionBumpMessage { get; internal set; }
 
     [JsonPropertyName("patch-version-bump-message")]
-    [JsonPropertyDescription(@"The regex to match commit messages with to perform a patch version increment. Default set to '\+semver:\s?(fix|patch)'")]
+    [JsonPropertyDescription(@"The regular expression to match commit messages with to perform a patch version increment. Default set to '\+semver:\s?(fix|patch)'")]
     [JsonPropertyPattern(@"'\+semver:\s?(fix|patch)'")]
-    public string? PatchVersionBumpMessage { get; set; }
+    public string? PatchVersionBumpMessage { get; internal set; }
 
     [JsonPropertyName("no-bump-message")]
     [JsonPropertyDescription(@"Used to tell GitVersion not to increment when in Mainline development mode. . Default set to '\+semver:\s?(none|skip)'")]
     [JsonPropertyPattern(@"'\+semver:\s?(none|skip)'")]
-    public string? NoBumpMessage { get; set; }
+    public string? NoBumpMessage { get; internal set; }
 
     [JsonPropertyName("label-pre-release-weight")]
     [JsonPropertyDescription("The pre-release weight in case of tagged commits. Defaults to 60000.")]
-    public int? LabelPreReleaseWeight { get; set; }
+    public int? LabelPreReleaseWeight { get; internal set; }
 
     [JsonPropertyName("commit-date-format")]
     [JsonPropertyDescription("The format to use when calculating the commit date. Defaults to 'yyyy-MM-dd'.")]
     [JsonPropertyPattern("'yyyy-MM-dd'", PatternFormat.DateTime)]
-    public string? CommitDateFormat { get; set; }
+    public string? CommitDateFormat { get; internal set; }
 
     [JsonPropertyName("merge-message-formats")]
     [JsonPropertyDescription("Custom merge message formats to enable identification of merge messages that do not follow the built-in conventions.")]
-    public Dictionary<string, string> MergeMessageFormats { get; set; } = new();
+    public Dictionary<string, string> MergeMessageFormats { get; internal set; } = new();
+
+    [JsonIgnore]
+    IReadOnlyDictionary<string, string> IGitVersionConfiguration.MergeMessageFormats => MergeMessageFormats;
 
     [JsonPropertyName("update-build-number")]
     [JsonPropertyDescription("Whether to update the build number in the project file. Defaults to true.")]
-    public bool UpdateBuildNumber { get; set; } = true;
+    public bool UpdateBuildNumber { get; internal set; } = true;
 
     [JsonPropertyName("semantic-version-format")]
     [JsonPropertyDescription("Specifies the semantic version format that is used when parsing the string. Can be 'Strict' or 'Loose'.")]
-    public SemanticVersionFormat SemanticVersionFormat { get; set; }
+    public SemanticVersionFormat SemanticVersionFormat { get; internal set; }
+
+    [JsonIgnore]
+    IReadOnlyDictionary<string, BranchConfiguration> IGitVersionConfiguration.Branches => Branches;
```

Comment:
I wonder if we can use the newly interface here
```suggestion
    IReadOnlyDictionary<string, IBranchConfiguration> IGitVersionConfiguration.Branches => Branches;
```
and the next line as well?

Reply:
Only the first line because the secon line is used for initialiization.